### ISSUE TITLE
Add extra scientific python package dependencies

### ISSUE THREAD

**Issue by jngrad** [espressomd/espresso#3933]
Wish list based on the summer school. Feel free to expand it.

* [`tidynamics`](https://joss.theoj.org/papers/10.21105/joss.00877)
   * offers: lightweight package for autocorrelation functions
   * why: avoids custom re-implementation of autocorrelation functions
   * where: only in the docker image for tutorials and samples
   * how: installed via pip
* [`pandas`](https://pandas.pydata.org/docs/getting_started/overview.html)
   * offers: data tables
   * why: help label datasets in tutorials (could it also help clarify code in the python test suite?)
   * where: in the image used for tutorials and samples, optionally in all images
   * how: Ubuntu package `python3-pandas`
* [`sympy`](https://www.sympy.org/en/index.html)
   * offers: limit/differentiation/integration calculus and a solver for differential equations
   * why: maybe useful for the reaction ensemble?
   * where: only in the image used for tutorials and samples
   * how: Ubuntu package `python3-sympy`

**Comment by jngrad** [comment]
* [`uncertainties`](https://uncertainties-python-package.readthedocs.io/en/latest/index.html)
   * offers: automatic error propagation, pretty printing, integrates into `pint`
   * why: avoids mistakes in error analysis
   * where: only in the docker image for tutorials and samples
   * how: Ubuntu package `python3-uncertainties`
   * example:
     ```python
     import numpy as np
     from pint import UnitRegistry
     ureg = UnitRegistry()
     n = (1e-3 * ureg.mole).plus_minus(1e-4)
     l = (1 * ureg.cm).plus_minus(0.03)
     c = n / l**3
     print(f'manual:    C = {c.magnitude.n} +/- {c.magnitude.n * np.sqrt((n.magnitude.s / n.magnitude.n)**2 + (3 * l.magnitude.s / l.magnitude.n)**2):.5f}')
     print(f'automatic: C = {c.magnitude}')
     ```
     ```
     manual:    C = 0.001 +/- 0.00013
     automatic: C = 0.00100+/-0.00013
     ```

**Comment by jngrad** [comment]
The core team decided to not move forward with this ticket, in an effort to limit the number of non-essential ESPResSo dependencies. The autocorrelation algorithm implemented in `tidynamics` was re-implemented with `scipy` in 662202a6029e492dde04ff816a90f2babfd56439.
